vars/handlePipelineStepErrors.groovy

```diff
@@ -1,34 +1,73 @@
+import com.cloudbees.groovy.cps.NonCPS
+
+import com.sap.piper.ConfigurationHelper
+
 import groovy.text.SimpleTemplateEngine
 import groovy.transform.Field
 
 @Field STEP_NAME = getClass().getName()
 
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STEP_CONFIG_KEYS = []
+@Field Set PARAMETER_KEYS = [
+    'echoDetails',
+    'libraryDocumentationUrl',
+    'libraryRepositoryUrl',
+    'stepName',
+    'stepNameDoc',
+    'stepParameters'
+]
+
 void call(Map parameters = [:], body) {
-    def stepParameters = parameters.stepParameters //mandatory
-    def stepName = parameters.stepName //mandatory
-    def verbose = parameters.get('echoDetails', true)
+    // load default & individual configuration
+    Map config = ConfigurationHelper.newInstance(this)
+        .loadStepDefaults()
+        .mixin(parameters, PARAMETER_KEYS)
+        .withMandatoryProperty('stepParameters')
+        .withMandatoryProperty('stepName')
+        .addIfEmpty('stepNameDoc' , parameters.stepName)
+        .use()
+
     def message = ''
     try {
-        if (stepParameters == null && stepName == null)
-            error "The step handlePipelineStepErrors requires following mandatory parameters: stepParameters, stepName"
-
-        if (verbose)
-            echo "--- Begin library step of: ${stepName} ---"
+        if (config.echoDetails)
+            echo "--- Begin library step of: ${config.stepName} ---"
 
         body()
-    } catch (Throwable err) {
-        if (verbose)
-            message += SimpleTemplateEngine.newInstance()
-                .createTemplate(libraryResource('com.sap.piper/templates/error.log'))
-                .make([
-                    stepName: stepName,
-                    stepParameters: stepParameters?.toString(),
-                    error: err
-                ]).toString()
-        throw err
+    } catch (Throwable error) {
+        if (config.echoDetails)
+            message += formatErrorMessage(config, error)
+        writeErrorToInfluxData(config, error)
```

Comment:
writing to influx is somehow a surprise when I have a look on the topic of this pr (useConfigurationHelper)